optimum/neuron/distributed/base.py

```diff
@@ -525,7 +529,48 @@ def load_model_checkpoint(cls, model: "PreTrainedModel", load_dir: Union[str, Pa
 
         if (load_dir / TENSOR_PARALLEL_SHARDS_DIR_NAME).is_dir():
             cls.load_model_sharded_checkpoint(model, load_dir)
-        elif (load_dir / WEIGHTS_NAME).is_file():
-            cls.load_model_regular_checkpoint(model, load_dir)
         else:
-            raise FileNotFoundError(f"Could not find a checkpoint file under {load_dir.as_posix()}.")
+            raise FileNotFoundError(f"Could not find a sharded checkpoint directory under {load_dir.as_posix()}.")
+
+    @classmethod
+    def load_optimizer_sharded_checkpoint(cls, optimizer: "torch.optim.Optimizer", load_dir: Union[str, Path]):
+        from neuronx_distributed.optimizer import NeuronZero1Optimizer
+
+        is_zero_1_optimizer = optimizer.__class__.__name__ == "NeuronAcceleratedOptimizer" and isinstance(optimizer.optimizer, NeuronZero1Optimizer)
+        is_zero_1_optimizer = is_zero_1_optimizer or isinstance(optimizer, NeuronZero1Optimizer)
```

Comment:
Isn't the check in the first line equivalent to `isinstance(optimizer, NeuronAcceleratedOptimizer)` ?
Anyway, since in the second line you also accept just `NeuronZero1Optimizer`, maybe it could be simplified to:
`is_zero_1_optimizer = isinstance(optimizer, NeuronZero1Optimizer`)

Reply:
No because you can have:

1. A `NeuronZero1Optimizer`, which we do not support yet
2. A `NeuronAcceleratedOptimizer` which is a wrapper around the optimizer that is always applied when using `accelerate` (I think we will always fall in that case but I wanted to cover all the cases here). This wrapper contains an `optimizer` attribute, and we check if it's a `NeuronZero1Optimizer`.